addons/purchase_stock/wizard/purchase_order_suggest.py

```diff
@@ -119,9 +131,13 @@ def action_purchase_order_suggest(self):
             # lines except the first one (who will be updated.)
             for po_line in existing_po_lines[1:]:
                 po_lines_commands.append(Command.unlink(po_line.id))
-            quantity = ceil(product.monthly_demand * self.multiplier)
-            qty_to_deduce = max(product.qty_available, 0) + max(product.incoming_qty, 0)
-            quantity -= qty_to_deduce
+
+            if self.based_on == 'actual_demand':
+                quantity = product_qty_dict.get(product.id, 0)
+            else:
+                quantity = ceil(product.monthly_demand * self.multiplier)
+                qty_to_deduce = max(product.qty_available, 0) + max(product.incoming_qty, 0)
+                quantity -= qty_to_deduce
```

Comment:
Suggested change based on the other comment:
```suggestion
quantity = product.actual_demand if self.based_on == 'actual_demand' else ceil(product.monthly_demand * self.multiplier)
quantity_to_reduce = max(product.qty_available, 0) + max(product.incoming_qty, 0) if self.based_on != 'actual_demand' else 0
quantity -= qty_to_reduce
```

Reply:
@mohammadhossam 
As we couldn't entirely get rid of the if-else branching, the if-else blocks looks more intuitive and logical to me (than the conditional operator) as I don't calculate any `quantity_to_deduce` if it's based on `actual_demand` but I don't know if there is a PYTHON way to write it.